### ISSUE TITLE
Support out-of-bounds predication for the `cp.async` for GMEM→SMEM (pre-Hopper)

### ISSUE THREAD

**Issue by samanklesaria** [jax-ml/jax#40048]
On pre-Hopper GPUs, `copy_gmem_to_smem` lowers to `cp.async` instead of TMA, and that path rejects every `oob_mode` except `PROMISE_IN_BOUNDS`:

```python
# jax/_src/pallas/mosaic_gpu/primitives.py:997
# cp.async does not predicate out-of-bounds accesses, so the caller has to
# guarantee that the copy stays in bounds.
if oob_mode != OOBFillMode.PROMISE_IN_BOUNDS:
    raise ValueError(
        "The cp.async implementation only supports "
        "oob_mode=OOBFillMode.PROMISE_IN_BOUNDS"
    )
```

`plgpu.emit_pipeline` inherits the restriction, so on Ampere every input block must tile its axis *exactly*:

```python
# jax/_src/pallas/mosaic_gpu/pipeline.py:351
for spec in effective_in_specs:
    if spec.oob_fill_mode is not OOBFillMode.PROMISE_IN_BOUNDS:
        raise NotImplementedError(
            "emit_pipeline on pre-Hopper GPUs requires input accesses to be"
            " provably in bounds, or oob_fill_mode="
            "OOBFillMode.PROMISE_IN_BOUNDS to be set explicitly"
        )
```

This is a gap in the Mosaic lowering, not a hardware limitation — `cp.async` supports both guard predication and hardware zero-fill.

### What I'd like to work

A pipelined elementwise kernel whose input block does not divide the operand. This works on Hopper today and raises on Ampere.

```python
import jax
import jax.numpy as jnp
from jax.experimental import pallas as pl
from jax.experimental.pallas import mosaic_gpu as plgpu

BLOCK_ROWS, COLS, GRID = 2, 128, 2
# 3 rows over blocks of 2: the second block reads one row past the end.
x = jnp.arange(3 * COLS, dtype=jnp.float32).reshape(3, COLS)

def kernel(x_gmem, y_gmem):
    def compute(step, x_smem):
        (i,) = step
        y_gmem[pl.ds(i * BLOCK_ROWS, BLOCK_ROWS)] = x_smem[...] * 2.0

    plgpu.emit_pipeline(
        compute,
        grid=(GRID,),
        in_specs=[
            plgpu.BlockSpec(
                (BLOCK_ROWS, COLS),
                lambda i: (i, 0),
                oob_fill_mode=plgpu.OOBFillMode.ZEROS,
            )
        ],
        max_concurrent_steps=2,
    )(x_gmem)


f = plgpu.kernel(
    kernel,
    out_type=jax.ShapeDtypeStruct((GRID * BLOCK_ROWS, COLS), x.dtype),
)
y = f(x)
# Expected with ZEROS fill: rows 0-2 are 2*x, row 3 is the zero-filled tail.
assert (y[:3] == x * 2.0).all()
assert (y[3] == 0.0).all()
```
